packages/editor/src/components/post-actions/actions.js

```diff
@@ -21,19 +23,46 @@ export function usePostActions( { postType, onActionPerformed, context } ) {
 		[ postType ]
 	);
 
+	const { canManageOptions, hasFrontPageTemplate } = useSelect(
+		( select ) => {
+			const { getEntityRecords } = select( coreStore );
+			const templates = getEntityRecords( 'postType', 'wp_template', {
+				per_page: -1,
+			} );
+
+			return {
+				canManageOptions: select( coreStore ).canUser( 'update', {
+					kind: 'root',
+					name: 'site',
+				} ),
+				hasFrontPageTemplate: !! templates?.find(
+					( template ) => template?.slug === 'front-page'
+				),
+			};
+		}
+	);
+
+	const setAsHomepageAction = useSetAsHomepageAction();
+	const shouldShowSetAsHomepageAction =
+		canManageOptions || ! hasFrontPageTemplate;
+
 	const { registerPostTypeActions } = unlock( useDispatch( editorStore ) );
 	useEffect( () => {
 		registerPostTypeActions( postType );
 	}, [ registerPostTypeActions, postType ] );
 
 	return useMemo( () => {
+		let actions = [
+			shouldShowSetAsHomepageAction ? setAsHomepageAction : [],
+			...defaultActions,
+		];
```

Comment:
```suggestion
		let actions = [
			...defaultActions,
			shouldShowSetAsHomepageAction ? setAsHomepageAction : []
		];
```

If we're _extending_ the defaults then the defaults should come first in the spread.

Reply:
Makes sense, updated the order in https://github.com/WordPress/gutenberg/pull/65426/commits/86af09abd2d5726caecaa69522c67fb16e4258b6.